queries/distance.py

```diff
@@ -147,27 +154,14 @@
 
 def _addr2nom(address):
     """ Convert location name to nominative form. """
-    # TODO: Implement more intelligently.
-    # This is a tad simplistic and mucks up some things,
-    # e.g. "Ráðhús Reykjavíkur" becomes "Ráðhús Reykjavík".
-    with BIN_Db.get_db() as db:
-        nf = []
-        for w in address.split():
-            bin_res = db.lookup_nominative(w)
-            if not bin_res and not w.islower():
-                # Try lowercase form
-                bin_res = db.lookup_nominative(w.lower())
-            if bin_res:
-                nf.append(bin_res[0].ordmynd)
-            else:
-                nf.append(w)
-        return " ".join(nf)
+    return NounPhrase(cap_first(address)).nominative or address
```

Comment:
Probably need to check for exception here, and also for empty address (None or "")